custom_components/hafas/sensor.py

```diff
@@ -84,88 +85,56 @@ def __init__(
         self._attr_name = title
         self._attr_icon = ICON
         self._attr_unique_id = entry_id
+        self._attr_device_class = SensorDeviceClass.TIMESTAMP
         self._attr_attribution = "Provided by " + profile + " through HaFAS API"
 
         self.journeys: list[Journey] = []
 
     def calc_native_value(self) -> str:
         """Return the departure time of the next train."""
-        if (
-            len(self.journeys) == 0
-            or self.journeys[0].legs is None
-            or len(self.journeys[0].legs) == 0
-        ):
-            return "No connection possible"
+        if not self.journeys:
+            return None
 
-        first_leg = self.journeys[0].legs[0]
+        connections = to_dict(self.journeys)
+        # use get method, because an empty Journey would return {}
+        running = [x for x in connections if not x.get("canceled", True)]
 
-        value = first_leg.departure.strftime("%H:%M")
-        if (
-            first_leg.departureDelay is not None
-            and first_leg.departureDelay != timedelta()
-        ):
-            delay = int(first_leg.departureDelay.total_seconds() // 60)
-
-            value += f" + {delay}"
+        if len(running) > 0:
+            return running[0]["departure"] + (
+                dt_util.parse_duration(running[0]["delay"]) or timedelta()
+            )
 
-        return value
+        return None
 
     def calc_extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes."""
-        if (
-            len(self.journeys) == 0
-            or self.journeys[0].legs is None
-            or len(self.journeys[0].legs) == 0
-        ):
+        attributes = {}
+
+        if not self.journeys:
             return {}
 
-        journey = self.journeys[0]
-        first_leg = journey.legs[0]
-        last_leg = journey.legs[-1]
-        products = ", ".join([x.name for x in journey.legs if x.name is not None])[:-2]
-        duration = timedelta() if journey.duration is None else journey.duration
-        delay = (
-            timedelta()
-            if first_leg.departureDelay is None
-            else first_leg.departureDelay
-        )
-        delay_arrival = (
-            timedelta() if last_leg.arrivalDelay is None else last_leg.arrivalDelay
-        )
+        connections = to_dict(self.journeys)
+        # use get method, because an empty Journey would return {}
+        running = [x for x in connections if not x.get("canceled", True)]
+
+        if len(running) > 0:
+            # use decomposition to not modify the original object
+            attributes = {k: v for k, v in running[0].items() if k != "legs"}
+
+        attributes["next"] = None
+        if len(running) > 1:
+            attributes["next"] = running[1]["departure"] + (
+                dt_util.parse_duration(running[1]["delay"]) or timedelta()
+            )
+
+        attributes["next_on"] = None
```

Comment:
see my suggestion about `next_on` in the general discussion of PR #4

Reply:
I removed the two attributes